crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -103,7 +103,10 @@ def dump_openstack_database
       node.save
     end
 
-    raise "There does not seem to exist any node running the database." if db_nodes.empty?
+    # Check if database proposal exists in case when somebody is trying to upgrade undeployed nodes
+    # or only with Ceph cluster
+    db_proposal = Proposal.where(barclamp: "database", name: "default").first
```

Comment:
Can you move that line outside of the loop? Also, please don't look for a proposal with name `default` as the name could be different; just look for any database proposal.